[iOS] Remove the image analysis timeout gesture

whsieh opened the pull request:
#### 84ccf0d321e00273f10ac00bf94d8d11d416bb76
<pre>
[iOS] Remove the image analysis timeout gesture
<a href="https://bugs.webkit.org/show_bug.cgi?id=264289">https://bugs.webkit.org/show_bug.cgi?id=264289</a>

Reviewed by NOBODY (OOPS!).

Remove the image analysis timeout gesture, along with the `disallowUserAgentShadowContent` position
information request flag (which was only set in this scenario). The timeout gesture currently allows
the user to show an image context menu after selecting Live Text, by continuing to hold down over
the image for 2 seconds after making a selection.

This was originally devised to make it possible to invoke the context menu over an image that is
completely covered by selectable Live Text; however:

1.  This scenario is extremely rare, and doesn&apos;t come up in practice; screenshots filled with text
    generally still contain enough space before or after lines, or in between paragraphs that it&apos;s
    possible to invoke the image context menu.

2.  The very long, 2 second delay made this fallback mechanism very difficult to discover in the
    first place.

For consistency with Live Text on the rest of the platform (e.g. Photos), and because it&apos;ll soon
become untennable to programmatically present context menus outside of the context of UI controls,
let&apos;s experiment with removing this fallback mechanism.

* Source/WebKit/Shared/ios/InteractionInformationRequest.cpp:
(WebKit::InteractionInformationRequest::isValidForRequest const):
* Source/WebKit/Shared/ios/InteractionInformationRequest.h:
(WebKit::InteractionInformationRequest::InteractionInformationRequest):
* Source/WebKit/Shared/ios/InteractionInformationRequest.serialization.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView setUpInteraction]):
(-[WKContentView cleanUpInteraction]):
(-[WKContentView gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:]):
(-[WKContentView gestureRecognizer:shouldRequireFailureOfGestureRecognizer:]):
(-[WKContentView gestureRecognizer:shouldBeRequiredToFailByGestureRecognizer:]):
(-[WKContentView deferringGestureRecognizer:shouldDeferOtherGestureRecognizer:]):
(-[WKContentView _setUpImageAnalysis]):
(-[WKContentView _tearDownImageAnalysis]):
(-[WKContentView _internalContextMenuInteraction:configurationForMenuAtLocation:completion:]):
(-[WKContentView imageAnalysisGestureDidTimeOut:]): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Drive-by fix: add `InteractionInformationRequest.serialization.in` to the Xcode project.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::positionInformation):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/84ccf0d321e00273f10ac00bf94d8d11d416bb76

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25102 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3643 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26358 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27217 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23038 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5347 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1079 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25345 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2670 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21664 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27796 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2362 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28730 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22901 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22954 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26544 "Passed tests") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2319 "Failed to checkout and rebase branch from PR 20063") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/613 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3661 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6015 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2754 "Built successfully") | | [⏳ 🛠 jsc-mips ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2650 "Built successfully") | | [⏳ 🧪 jsc-mips-tests ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
> The very long, 2 second delay made this fallback mechanism very difficult to discover in the first place.

Truly, because TIL about this UI.

whsieh committed:
[iOS] Remove the image analysis timeout gesture
https://bugs.webkit.org/show_bug.cgi?id=264289

Reviewed by Megan Gardner.

Remove the image analysis timeout gesture, along with the `disallowUserAgentShadowContent` position
information request flag (which was only set in this scenario). The timeout gesture currently allows
the user to show an image context menu after selecting Live Text, by continuing to hold down over
the image for 2 seconds after making a selection.

This was originally devised to make it possible to invoke the context menu over an image that is
completely covered by selectable Live Text; however:

1.  This scenario is extremely rare, and doesn't come up in practice; screenshots filled with text
    generally still contain enough space before or after lines, or in between paragraphs that it's
    possible to invoke the image context menu.

2.  The very long, 2 second delay made this fallback mechanism very difficult to discover in the
    first place.

For consistency with Live Text on the rest of the platform (e.g. Photos), and because it'll soon
become untennable to programmatically present context menus outside of the context of UI controls,
let's experiment with removing this fallback mechanism.

* Source/WebKit/Shared/ios/InteractionInformationRequest.cpp:
(WebKit::InteractionInformationRequest::isValidForRequest const):
* Source/WebKit/Shared/ios/InteractionInformationRequest.h:
(WebKit::InteractionInformationRequest::InteractionInformationRequest):
* Source/WebKit/Shared/ios/InteractionInformationRequest.serialization.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView setUpInteraction]):
(-[WKContentView cleanUpInteraction]):
(-[WKContentView gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:]):
(-[WKContentView gestureRecognizer:shouldRequireFailureOfGestureRecognizer:]):
(-[WKContentView gestureRecognizer:shouldBeRequiredToFailByGestureRecognizer:]):
(-[WKContentView deferringGestureRecognizer:shouldDeferOtherGestureRecognizer:]):
(-[WKContentView _setUpImageAnalysis]):
(-[WKContentView _tearDownImageAnalysis]):
(-[WKContentView _internalContextMenuInteraction:configurationForMenuAtLocation:completion:]):
(-[WKContentView imageAnalysisGestureDidTimeOut:]): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Drive-by fix: add `InteractionInformationRequest.serialization.in` to the Xcode project.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::positionInformation):

Canonical link: https://commits.webkit.org/270304@main
